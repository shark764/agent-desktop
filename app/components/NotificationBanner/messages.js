/*
 * NotificationBanner Messages
 *
 * This contains all the text for the NotificationBanner component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  tryAgain: {
    id: 'app.components.NotificationBanner.tryAgain',
    defaultMessage: 'Try again.',
  },
  serverError: {
    id: 'app.components.NotificationBanner.serverError',
    defaultMessage: 'Server Error',
  },
  networkError: {
    id: 'app.components.NotificationBanner.networkError',
    defaultMessage: 'Network Error',
  },
  created: {
    id: 'app.components.NotificationBanner.created',
    defaultMessage: 'Contact created.',
  },
  saved: {
    id: 'app.components.NotificationBanner.saved',
    defaultMessage: 'Changes saved.',
  },
  notSaved: {
    id: 'app.components.NotificationBanner.notSaved',
    defaultMessage: 'Your changes have not been saved.',
  },
  notCreated: {
    id: 'app.components.NotificationBanner.notCreated',
    defaultMessage: 'Your new contact has not been created.',
  },
  notAssigned: {
    id: 'app.components.NotificationBanner.notAssigned',
    defaultMessage: 'Contact was not assigned to the interaction.',
  },
});
