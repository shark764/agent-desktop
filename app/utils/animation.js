/**
 * throttle max function call rate to browser refresh rate
 */

const requestAnimationFrame =
  window.requestAnimationFrame ||
  window.webkitRequestAnimationFrame ||
  window.mozRequestAnimationFrame ||
  ((callback) => window.setTimeout(callback, 1000 / 60));

function throttler(...callbackArgs) {
  if (!this.running) {
    this.running = true;
    requestAnimationFrame(() => {
      this.callback(...callbackArgs);
      this.running = false;
    });
  }
}

export function throttleDecorator(funcToThrottle) {
  return throttler.bind({ callback: funcToThrottle, running: false });
}
