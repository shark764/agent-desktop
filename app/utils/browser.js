/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

// Check if browser used is IE11 or greater
// Returns undefined for Chrome|Firefox|Edge
export const isIeEleven = () => document.documentMode && document.documentMode >= 11;
