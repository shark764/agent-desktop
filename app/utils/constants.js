/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

export const colors = {
  lightGray: '#ebebeb',
  offWhite: '#e4e4e4',
  lightRed: '#FE4565',
  regularRed: '#E43D5A',
  darkRed: '#CB3750',
  darkGreen: '#001b1e',
  hoverDarkGreen: 'rgba(0, 27, 30, 0.85)',
  black: '#000',
  white: '#FFFFFF',
  skylightBlue: '#23cdf4',
  toggleChecked: '#54B84F',
  toggleCheckedDisabled: '#b7e3b3',
};
