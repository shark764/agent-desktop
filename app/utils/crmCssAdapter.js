/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
  * crmCssAdapter
  *
  * Updates CSS as needed to work with various CRM integrations
  *
  */

export default function crmCssAdapter (styleObject, styleNameArray, crmModule) {
  let styleAffix;
  if (crmModule === 'salesforce-lightning') {
    styleAffix = 'SfLightning';
  }

  styleNameArray.forEach((styleName) => {
    Object.assign(
      styleObject[styleName],
      styleObject[`${styleName}${styleAffix}`]
    );
  });
}