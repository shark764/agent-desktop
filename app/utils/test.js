import { IntlProvider } from 'react-intl';
import enUS from 'translations/en-US.json';

export const getIntlContext = () => {
  // Construct a new `IntlProvider` instance by passing `props` and
  // `context` as React would, then call `getChildContext()` to get the
  // React Intl API, complete with the `format*()` functions.
  const intlProvider = new IntlProvider({ locale: 'en-US', messages: enUS }, {});
  const { intl } = intlProvider.getChildContext();
  return intl;
};

export default { getIntlContext };
