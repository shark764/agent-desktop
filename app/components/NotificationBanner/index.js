/**
*
* NotificationBanner
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import Radium from 'radium';
import { injectIntl } from 'react-intl';

import Button from 'components/Button';

import messages from './messages';

function NotificationBanner(props) {
  const styles = {
    base: {
      backgroundColor: props.isError ? '#FE4565' : '#072931',
      width: 'calc(100% + 76px)',
      height: '35px',
      position: 'relative',
      left: '-51px',
      display: 'flex',
      alignItems: 'stretch',
      color: 'white',
      flexShrink: 0,
    },
    typeText: {
      fontWeight: '600',
      alignSelf: 'center',
    },
    messageText: {
      alignSelf: 'center',
    },
    tryAgainText: {
      textDecoration: 'underline',
      cursor: 'pointer',
      alignSelf: 'center',
    },
    leftGutter: {
      width: '52px',
    },
    closeButton: {
      margin: '0 10px 0 0',
      border: '0',
      alignSelf: 'center',
    },
  };

  return (
    <div id={props.id} style={[styles.base]} key={props.key}>
      <div style={styles.leftGutter}></div>
      {
        props.isError
        ? <div key="1" style={styles.typeText}>
          {props.intl.formatMessage(messages[`${props.errorType}`])}
          &nbsp;
        </div>
        : undefined
      }
      <div style={styles.messageText}>
        {props.intl.formatMessage(messages[props.messageType])}
      </div>
      {
        props.tryAgain ? <div onClick={props.tryAgain} style={styles.tryAgainText}>
          {props.intl.formatMessage(messages.tryAgain)}
        </div>
        : undefined
      }
      <div style={{ flexGrow: 1 }}></div>
      {
        props.isError
        ? <Button
          id={`${props.id}error-dismiss-btn`}
          style={styles.closeButton}
          iconName="close"
          type="primaryRed"
          onClick={props.dismiss}
        />
        : undefined
      }
    </div>
  );
}

NotificationBanner.propTypes = {
  intl: PropTypes.object.isRequired,
  id: PropTypes.string.isRequired,
  errorType: PropTypes.oneOf(['serverError', 'networkError']),
  messageType: PropTypes.oneOf(['notSaved', 'notCreated', 'notAssigned', 'created', 'saved']).isRequired,
  dismiss: PropTypes.func.isRequired,
  isError: PropTypes.bool.isRequired,
  tryAgain: PropTypes.func,
  key: PropTypes.string,
};

export default injectIntl(Radium(NotificationBanner));
