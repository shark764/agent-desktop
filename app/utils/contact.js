/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { isValidEmail, isPossibleNumber } from 'utils/validator';
import isURL from 'validator/lib/isURL';

export const formatValue = (attributeToValidate, value) => {
  let formattedValue;
  switch (attributeToValidate.type) {
    case 'phone':
      formattedValue = value.replace(/[^0-9+*#]/g, '');
      if (formattedValue.indexOf('+') !== 0 && formattedValue.length > 0) {
        formattedValue = `+${formattedValue}`;
      }
      break;
    case 'boolean':
      if (value === 'false' || value === '') formattedValue = false;
      else formattedValue = !!value;
      break;
    default:
      formattedValue = value;
  }
  return formattedValue;
};

export const getError = (attributeToValidate, value) => {
  let error = false;
  if (
    attributeToValidate.mandatory &&
    attributeToValidate.type !== 'boolean' &&
    (value === undefined || value.trim().length < 1)
  ) {
    error = 'errorRequired';
  } else if (value !== undefined && value.length) {
    switch (attributeToValidate.type) {
      case 'email':
        if (!isValidEmail(value)) {
          error = 'errorEmail';
        }
        break;
      case 'phone':
        try {
          if (!isPossibleNumber(value)) {
            error = 'errorPhone';
          }
        } catch (e) {
          error = 'errorPhone';
        }
        break;
      case 'link':
        if (
          !isURL(value, {
            protocols: ['http', 'https'],
            require_protocol: true,
          })
        ) {
          error = 'errorLink';
        }
        break;
      case 'number':
        if (Number.isNaN(Number(value))) {
          error = 'errorNumber';
        }
        break;
      default:
        break;
    }
  }
  return error;
};

export const getLocaleLabel = (attribute, locale) => {
  if (attribute.label[locale] !== undefined) {
    return attribute.label[locale];
  } else if (attribute.label.all !== undefined) {
    // Special case for 'All'. We have our own local translation for it we use.
    return attribute.label.all;
  } else if (attribute.objectName !== undefined) {
    // Fallback for attributes
    return attribute.objectName;
  } else {
    // Fallback for section headers
    return '';
  }
};

export function createSearchQuery(query, nextPage) {
  const encodedQuery = {};
  Object.keys(query).forEach(queryName => {
    const queryToEncode = query[queryName];
    encodedQuery[queryName] = encodeURIComponent(queryToEncode);
  });
  return {
    query: Object.assign(encodedQuery, {
      page: nextPage !== undefined ? nextPage : 1,
    }),
  };
}
