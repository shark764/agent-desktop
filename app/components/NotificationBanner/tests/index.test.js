/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { getIntlContext } from 'utils/test';

import NotificationBanner from '../index';

const mockMessage = { id: 'app.test.mockMessage', defaultMessage: 'mockMessage' };

describe('<NotificationBanner />', () => {
  it('should render correctly', () => {
    const rendered = shallow(
      <NotificationBanner.WrappedComponent
        id="mockId"
        titleMessage={mockMessage}
        descriptionMessage={mockMessage}
        dismiss={() => {}}
        isError
        intl={getIntlContext()}
      />
    );
    expect(rendered).toMatchSnapshot();
  });
});
