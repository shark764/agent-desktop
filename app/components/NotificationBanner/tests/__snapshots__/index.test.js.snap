// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBanner /> should render correctly 1`] = `
<div
  data-radium={true}
  id="mockId"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FE4565",
      "color": "white",
      "display": "flex",
      "flexShrink": 0,
      "height": "35px",
      "paddingLeft": "52px",
      "width": "100%",
    }
  }
>
  <div
    data-radium={true}
    style={
      Object {
        "fontWeight": "600",
      }
    }
  >
    Mock message
    Â 
  </div>
  <div>
    Mock message
  </div>
  <div
    data-radium={true}
    style={
      Object {
        "flexGrow": 1,
      }
    }
  />
  <Button
    iconName="close"
    id="mockIderror-dismiss-btn"
    onClick={[Function]}
    style={
      Object {
        "borderBottom": "0",
        "borderLeft": "0",
        "borderRight": "0",
        "borderTop": "0",
        "margin": "0 10px 0 0",
      }
    }
    tabIndex={0}
    type="primaryRed"
  />
</div>
`;
